core/storage/user/gae_models_test.py

```diff
@@ -27,43 +27,105 @@
 
 class UserSettingsModelTest(test_utils.GenericTestBase):
     """Tests for UserSettingsModel class."""
+    user_id = 'user_id'
     user_email = 'user@example.com'
     user_role = feconf.ROLE_ID_ADMIN
     user2_email = 'user2@example.com'
     user2_role = feconf.ROLE_ID_BANNED_USER
+    user3_email = 'user3@example.com'
+    user3_role = feconf.ROLE_ID_ADMIN
+    user3_id = 'user3_id'
+    generic_username = 'user'
+    generic_date = datetime.datetime(2019, 5, 20)
+    generic_image_url = 'www.example.com/example.png'
+    generic_user_bio = 'I am a user of Oppia!'
+    generic_subject_interests = ['Math', 'Science']
+    generic_language_codes = ['en', 'es']
 
     def setUp(self):
         super(UserSettingsModelTest, self).setUp()
         user_models.UserSettingsModel(
-            email=self.user_email, role=self.user_role).put()
+            id=self.user_id, email=self.user_email, role=self.user_role).put()
         user_models.UserSettingsModel(
             email=self.user2_email, role=self.user2_role).put()
+        user_models.UserSettingsModel(
+            email=self.user3_email, role=self.user3_role).put()
+        user_models.UserSettingsModel(
+            id=self.user3_id,
+            email=self.user3_email,
+            role=self.user3_role,
+            username=self.generic_username,
+            normalized_username=self.generic_username,
+            last_agreed_to_terms=self.generic_date,
+            last_started_state_editor_tutorial=self.generic_date,
+            last_started_state_translation_tutorial=self.generic_date,
+            last_logged_in=self.generic_date,
+            last_created_an_exploration=self.generic_date,
+            last_edited_an_exploration=self.generic_date,
+            profile_picture_data_url=self.generic_image_url,
+            default_dashboard='learner', creator_dashboard_display_pref='card',
+            user_bio=self.generic_user_bio,
+            subject_interests=self.generic_subject_interests,
+            first_contribution_msec=1,
+            preferred_language_codes=self.generic_language_codes,
+            preferred_site_language_code=(self.generic_language_codes[0]),
+            preferred_audio_language_code=(self.generic_language_codes[0])
+        ).put()
 
     def test_get_by_role(self):
         user = user_models.UserSettingsModel.get_by_role(
             feconf.ROLE_ID_ADMIN)
         self.assertEqual(user[0].role, feconf.ROLE_ID_ADMIN)
 
-
-class StoryProgressModelTests(test_utils.GenericTestBase):
```

Comment:
Please please **always** check the "Files Changed" tab very carefully immediately after you push, to make sure that the diff is what you expect it to be. You appear to have deleted an entire test class inadvertently here :(

Reply:
This is so weird. How does this happen? My commit itself has only the changes I intended to make, but when I pushed to varun's takeout version this test was lost in the merging process? That doesn't make any sense to me, I'm not sure how to prevent that from happening?

Reply:
You can probably figure out what went wrong by examining the [commit history](https://github.com/oppia/oppia/pull/6981/commits), but the mitigation really is to always check the "Files Changed" tab really carefully after submitting. That catches most things :)